### PR TITLE
Make loading of detailed measurements optional

### PR DESCRIPTION
**Pull request by ArneTR**
Loading of a complex detailed measurements page can take enormous amounts of time du to the large amount of data transferred and the long query in the database.

Since this value cannot really be brought down, it is already compressed and DB indices are set, this PR shall enable the option to make the data transfer optional.

Especially beginner users do not need the timeline charts and rather use the phase_stats.

See the timing from the developer console attached. The PR brings this down to 10%

Screenshot 2023-12-15 at 10 33 16 AM

### CODE REVIEW

**Comment by ArneTR** [comment]
Forgot to mention: This behaviour can also be changed in settings. By default loading data is off.

In every run this button will be shown:

Screenshot 2023-12-15 at 10 48 39 AM

File: frontend/js/stats.js [hunk 1]
```diff
@@ -355,36 +348,73 @@ const getURLParams = () => {
...
 
-        let [run_data, measurements_data, notes_data, phase_stats_data, network_data] = await makeAPICalls(url_params);
+        let [run_data, phase_stats_data, network_data] = await makeBaseAPICalls(url_params);
 
         if (run_data == undefined) return;
 
         renderBadges(url_params);
 
         fillRunData(run_data);
+        console.log(phase_stats_data);
```
Comment (ribalba) [inline frontend/js/stats.js:407]: Should the log stay in there?

File: frontend/js/stats.js [hunk 2]
```diff
@@ -355,36 +348,73 @@ const getURLParams = () => {
...
-        if (measurements_data == undefined) return;
-        const metrics = getTimelineMetrics(measurements_data);
-
-        if (notes_data == undefined) return;
-        displayTimelineCharts(metrics, notes_data);
+        if (localStorage.getItem('fetch_detailed_measurements') === 'true') {
+            getDetailedMeasurements(url_params);
+        }
 
-        displayNetworkIntercepts(network_data);
```
Comment (ribalba) [inline frontend/js/stats.js:387]: And this?

**Comment by ribalba** [comment]
I really like the idea. But wouldn't it be easier just to load the data if someone clicks on the "Click here for detailed metrics ..." button. You could show a little loading spinner. Like this people don't really need to know the details and it would just be in the flow of looking at things?

Screenshot 2023-12-15 at 10 51 27

**Comment by ArneTR** [comment]
Thanks for the spot on the outcommented code. A good reminder to always to a git diff.

2. The button "Click here for detailed metrics" contains a table and in comparison mode the comparison charts.

I see this as different data, although the naming is pretty close atm.

I am personally not a fan of clicking on a button and then data appears in two different places, so I would not like to integrate it that way.
Especially since the behaviour will be different between stats.html and compare.html. In the latter no detailed time series will appear at the bottom.

I would keep it very local. If you click on a button that talks about data, it should appear right below it.

Having said that: Tackling the loading issue for the compare.html is a separate task. I have planned that out already, but since it is not tied to fetch events, but phase calculations internally this is a bigger one

**Comment by ribalba** [comment]
Sorry I wasn't clear. I think it is confusing to have the same button twice. I wouldn't know what the difference is between the two and it is confusing that they are pretty much named the same. I would have one button for detailed metrics and not one which opens a table and one which loads graphs and then a setting that only sets the one. We would have to restructure the way the page is laid out though in this case.

**Comment by ArneTR** [comment]
as said, this is iteratively. The latter will come as a seprate functionality.

I agree on the naming though. Have pushed to main and call it "Detailed Time Series"

**Pull request merged**